app/inflation/inflation.go

```diff
@@ -47,24 +46,26 @@ func (c Calculator) InflationRate(ctx sdk.Context, minter minttypes.Minter, mint
 		)
 	}
 
-	minter.Inflation = minttypes.DefaultInflationCalculationFn(ctx, minter, mintParams, bondedRatio)
-	return c.adjustInflation(totalSupply, inflationParams.MaxSupply.Amount, minter, mintParams)
+	minter.Inflation = minter.NextInflationRate(mintParams, bondedRatio)
+	return c.AdjustInflation(totalSupply, inflationParams.MaxSupply.Amount, minter, mintParams)
 }
 
-// adjustInflation check if newly minting coins will execeed the MaxSupply then it will adjust the inflation with
+// AdjustInflation check if newly minting coins will execeed the MaxSupply then it will adjust the inflation with
 // respect to MaxSupply
-func (c Calculator) adjustInflation(totalSupply, maxSupply math.Int, minter minttypes.Minter,
+func (c Calculator) AdjustInflation(totalSupply, maxSupply math.Int, minter minttypes.Minter,
 	params minttypes.Params) sdk.Dec {
 	minter.AnnualProvisions = minter.NextAnnualProvisions(params, totalSupply)
 	newSupply := minter.BlockProvision(params).Amount
 	newTotalSupply := totalSupply.Add(newSupply)
+
 	if newTotalSupply.GT(maxSupply) {
 		newTotalSupply = maxSupply.Sub(totalSupply)
```

Comment:
It seems like outside of this block, `newTotalSupply` is defined as "the total supply, after new supply is added" but inside, it is defined as "how much the total supply is allowed to increase".

Consider using a new variable or commenting above the code so it is clear what is being done

Reply:
`how much the total supply is allowed to increase` is the MaxSupply Param and 
`newTotalSupply = minting Coins + stakingTokenSupply`